common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -652,6 +827,274 @@ class NativeAuthMsalController : BaseNativeAuthController() {
+            }
+            is SignUpContinueApiResult.InvalidAttributes -> {
+                SignUpCommandResult.InvalidAttributes(
+                    error = this.error,
+                    errorDescription = this.errorDescription,
+                    invalidAttributes = this.invalidAttributes
+                )
+            }
+
+            is SignUpContinueApiResult.InvalidOOBValue, is SignUpContinueApiResult.InvalidPassword,
+            is SignUpContinueApiResult.ExpiredToken, is SignUpContinueApiResult.UnknownError -> {
+                Logger.warn(
+                    TAG,
+                    "Expire token result: $this"
+                )
+                this as ApiErrorResult
+                INativeAuthCommandResult.UnknownError(
+                    error = this.error,
+                    errorDescription = this.errorDescription
+                )
+            }
+        }
+    }
+
+    private fun SignUpContinueApiResult.toSignUpSubmitPasswordCommandResult(
+        oAuth2Strategy: NativeAuthOAuth2Strategy
+    ): SignUpSubmitPasswordCommandResult {
+        return when (this) {
+            is SignUpContinueApiResult.Success -> {
+                SignUpCommandResult.Complete(
+                    signInSLT = this.signInSLT,
+                    expiresIn = this.expiresIn
+                )
+            }
+
+            is SignUpContinueApiResult.UsernameAlreadyExists -> {
+                SignUpCommandResult.UsernameAlreadyExists(
+                    error = this.error,
+                    errorDescription = this.errorDescription
+                )
+            }
+            is SignUpContinueApiResult.AttributesRequired -> {
+                SignUpCommandResult.AttributesRequired(
+                    signupToken = this.signupToken,
+                    error = this.error,
+                    errorDescription = this.errorDescription,
+                    requiredAttributes = this.requiredAttributes
+                )
+            }
+            is SignUpContinueApiResult.CredentialRequired -> {
+                return performSignUpChallengeCall(
+                    oAuth2Strategy = oAuth2Strategy,
+                    signupToken = this.signupToken
+                ).toSignUpStartCommandResult() as SignUpSubmitPasswordCommandResult
+            }
+            is SignUpContinueApiResult.InvalidPassword -> {
+                SignUpCommandResult.InvalidPassword(
+                    error = this.error,
+                    errorDescription = this.errorDescription
+                )
+            }
+            is SignUpContinueApiResult.Redirect -> {
+                INativeAuthCommandResult.Redirect()
+            }
+            is SignUpContinueApiResult.ExpiredToken, is SignUpContinueApiResult.InvalidOOBValue,
+            is SignUpContinueApiResult.InvalidAttributes, is SignUpContinueApiResult.UnknownError -> {
+                Logger.warn(
+                    TAG,
+                    "Expire token result: $this"
```

Comment:
Do we want to log this even when the error we receive is not ExpiredToken? We will log this even if we get InvalidOOBValue.

Would probably cause some confusion during debugging